UI/ViewModel/CategoryControllerViewModel.cs

```diff
@@ -103,12 +100,12 @@ public CategoryControllerViewModel(ICommandFactory commandFactory, ITodoService
         /// Update from serveice.
         /// </summary>
         /// <param name="model">Model. </param>
-        public void Update(ICategoryController model)
+        public void Refresh(ICategoryController model)
         {
             List.Clear();
             model.SelectAll()
-                .ToList()
-                .ForEach(item => CreateItem().Update(item));
+                .OrderBy(t => t.Order)
+                .ForEachEx(item => CreateItem().Refresh(item));
```

Comment:
Now I understand why you used ToList and this was actually better than ForEachEx extension. Please return ToList back). If I ask something you can just replay and explain why you made such desicion.